File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.swift
class Solution {
        func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
               // two pointer solution
            var L = 0
            var R = numbers.count - 1
            
            while L < R {
                if numbers[L] + numbers[R] > target{
                    R -= 1
                }else if numbers[L] + numbers[R] < target {
                    L += 1
                }else{
                    return [L+1, R+1]
                }
            }
            return [] 

        }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nDict = {}
        # one pass 
        for i, n in enumerate(nums):
            comp = target - n
            if comp in nDict: 
                return [nDict[comp], i]
            # populate dict 
            nDict[n] = i 
        return [] 

        